### ISSUE TITLE
polymorphicDefaultDeserializer is unbound error

### ISSUE THREAD

**Issue by tfonrouge** [pdvrieze/xmlutil#126]
Hello,

I'm having the following error when compiling with the multiplatform XML-serialization library:

```
e: java.lang.IllegalStateException: Symbol for kotlinx.serialization.modules/SerializersModuleCollector.polymorphicDefaultDeserializer|-5554727982800535415[0] is unbound
	at org.jetbrains.kotlin.ir.symbols.impl.IrBindablePublicSymbolBase.getOwner(IrPublicSymbolBase.kt:52)
	at org.jetbrains.kotlin.ir.backend.js.ic.CacheUpdater.collectImplementedSymbol(CacheUpdater.kt:299)
	at org.jetbrains.kotlin.ir.backend.js.ic.CacheUpdater.getExportedSignaturesAndAddMetadata-gNUL48g(CacheUpdater.kt:351)
	at org.jetbrains.kotlin.ir.backend.js.ic.CacheUpdater.rebuildDirtySourceMetadata(CacheUpdater.kt:403)
	at org.jetbrains.kotlin.ir.backend.js.ic.CacheUpdater.actualizeCachesImpl(CacheUpdater.kt:704)
	at org.jetbrains.kotlin.ir.backend.js.ic.CacheUpdater.actualizeCaches(CacheUpdater.kt:738)
	at org.jetbrains.kotlin.cli.js.K2JsIrCompiler.doExecute(K2JsIrCompiler.kt:300)
	at org.jetbrains.kotlin.cli.js.K2JSCompiler.doExecute(K2JSCompiler.java:183)
	at org.jetbrains.kotlin.cli.js.K2JSCompiler.doExecute(K2JSCompiler.java:72)
	at org.jetbrains.kotlin.cli.common.CLICompiler.execImpl(CLICompiler.kt:101)
	at org.jetbrains.kotlin.cli.common.CLICompiler.execImpl(CLICompiler.kt:47)
	at org.jetbrains.kotlin.cli.common.CLITool.exec(CLITool.kt:101)
	at org.jetbrains.kotlin.daemon.CompileServiceImpl.compile(CompileServiceImpl.kt:1645)
	at jdk.internal.reflect.GeneratedMethodAccessor101.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at java.rmi/sun.rmi.server.UnicastServerRef.dispatch(UnicastServerRef.java:360)
	at java.rmi/sun.rmi.transport.Transport$1.run(Transport.java:200)
	at java.rmi/sun.rmi.transport.Transport$1.run(Transport.java:197)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:712)
	at java.rmi/sun.rmi.transport.Transport.serviceCall(Transport.java:196)
	at java.rmi/sun.rmi.transport.tcp.TCPTransport.handleMessages(TCPTransport.java:587)
	at java.rmi/sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.run0(TCPTransport.java:828)
	at java.rmi/sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.lambda$run$0(TCPTransport.java:705)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.rmi/sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.run(TCPTransport.java:704)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)

```
The above error doesn't shows when using the JVM XML-serialization library, only appears with the multiplatform library version.

Seems to be related to the kotlinx-serialization library version, I have some dependencies that forces me to use the kotlinx-serialization 1.5.0.

Also, the following youtrack seems to be related to this issue:

https://youtrack.jetbrains.com/issue/KT-56602

thanks in advance for your support

**Comment by pdvrieze** [comment]
Thanks for the reference to the youtrack issue. I've managed to find and fix it. Will be released in 0.85.1 (and as snapshot).

**Comment by tfonrouge** [comment]
Hi,

I've managed building the SNAPSHOT and problem solved.

Thank you very much for your support !


Teo Fonrouge
